Gemfile

```diff
@@ -28,22 +28,14 @@ group :system_tests do
   gem 'beaker-puppet_install_helper', :require => false
 end
 
-is_x64 = Gem::Platform.local.cpu == 'x64'
-if is_x64
-  platform(:x64_mingw) do
-    gem "win32-dir", "~> 0.4.9", :require => false
-    gem "win32-process", "~> 0.7.4", :require => false
-    gem "win32-service", "~> 0.8.4", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-  end
-else
-  platform(:mingw) do
-    gem "win32-process", "~> 0.6.5", :require => false
-    gem "win32-service", "~> 0.7.2", :require => false
-    gem "minitar", "~> 0.5.4", :require => false
-    if RUBY_VERSION =~ /^1\./ then
-      gem "win32console", :require => false
-    end
+platforms :mswin, :mingw do
```

Comment:
Sorry @ferventcoder - I left a drive-by "me too" comment on here as I was rushing out the door, without offering up a lot of explanation.  I apologize if that seemed offensive, and I apologize for not being clear about the "dangerous" part (which would be using the `:x64_mingw` symbol without the current guard in place).  For some more background on that goofy guard, see:

https://github.com/puppetlabs/puppet/commit/f6eb2afe292902d1fccaeadb71425da9ee6d12be
https://github.com/puppetlabs/puppet/commit/b6a20930d589daadc36f7681fc40b1a0c6bb29c3

The problem is that `:mswin` doesn't actually match our `x64` Ruby runtime, and won't install platform specific gems in an `x64-mingw32` environment.  See [bundler code](https://github.com/bundler/bundler/blob/f4598e4a01d66677dbd54c9b1d8fd6784daf7271/lib/bundler/current_ruby.rb) for how the platforms are determined - it can be surprising.  (As you've noted, you're seeing the desired end result, but it's a bit of a coincide that it works - the `:mswin` part is meaningless)

You can simulate this with a quick test inside and x64 Ruby with this `Gemfile`.

```ruby
source "https://rubygems.org"

gem "ffi", "1.9.8", :platforms => [:mswin]
```

```
C:\source\x64-gem-test> ruby --version
ruby 2.1.5p273 (2014-11-13 revision 48405) [x64-mingw32]

C:\source\x64-gem-test> bundle install --path .bundle\gems
DL is deprecated, please use Fiddle
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Using bundler 1.7.12
Your bundle is complete!
It was installed into ./.bundle/gems

C:\source\x64-gem-test> type .\Gemfile.lock
GEM
  remote: https://rubygems.org/
  specs:

PLATFORMS
  x64-mingw32

DEPENDENCIES
  ffi (= 1.9.8)
```

So as you can see, the `ffi` gem doesn't actually get installed, because the Bundler platform is not a match.


To illustrate the correct behavior, you could change the `Gemfile`

```ruby
source "https://rubygems.org"

gem "ffi", "1.9.8", :platforms => [:x64_mingw]
```

Clear the lock, and .. voila, `ffi` is installed.

```
C:\source\x64-gem-test> del Gemfile.lock
C:\source\x64-gem-test> del .\.bundle\ -Recurse -Force
C:\source\x64-gem-test> bundle install --path .bundle\gems
DL is deprecated, please use Fiddle
Fetching gem metadata from https://rubygems.org/...........
Resolving dependencies...
Using bundler 1.7.12
Installing ffi 1.9.8
Your bundle is complete!
It was installed into ./.bundle/gems

C:\source\x64-gem-test> type Gemfile.lock
GEM
  remote: https://rubygems.org/
  specs:
    ffi (1.9.8-x64-mingw32)

PLATFORMS
  x64-mingw32

DEPENDENCIES
  ffi (= 1.9.8)
```

I agree that gem dependencies should be picked up automatically by the inclusion of `Puppet` itself (which means that we shouldn't have to include any win32 dependencies).  I think that you've discovered one of the reasons that this exists in the first place is Puppet 3.3.0.  I think another reason is that we didn't start shipping platform gems until `3.5.0` - see https://rubygems.org/gems/puppet/versions

I'm wondering if there was another historical reason that the `Gemfile` looks like this that I'm forgetting.  It might make sense to guard all of this nonsense based on Puppet gem version if that's the rationale.  I think being explicit there could make it much clearer to future maintainers.